localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_response.py

```diff
@@ -128,3 +149,59 @@ def select_integration_response(
                     selected_response["statusCode"],
                 )
         return selected_response
+
+    @staticmethod
+    def get_response_template(
+        integration_response: IntegrationResponse, request: InvocationRequest
+    ) -> str:
+        """The Response Template is selected from the response templates.
+        If there are no templates defined, the body will pass through.
+        Apigateway looks at the integration request `Accept` header and defaults to `application/json`.
+        If no template is matched, Apigateway will use the "first" existing template and use it as default.
+        https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html#transforming-request-response-body
+        """
+        if not (response_templates := integration_response["responseTemplates"]):
+            return ""
+
+        # Aws seems to ignore the integration request headers and uses the invocation request header
+        accept = request["headers"].get("accept", APPLICATION_JSON)
+        if template := response_templates.get(accept):
+            return template
+        # TODO aws seemed to favor application/json as default when unmatched regardless of "first"
+        if template := response_templates.get(APPLICATION_JSON):
+            return template
```

Comment:
Yeah, it is a bit redundant in the case where it isn't set and `application/json` does not exists. It might also be simpler to follow the logic without that redundancy, Good call